internal/provider/burn_alert_resource.go

```diff
@@ -97,10 +98,25 @@ func (*burnAlertResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 				},
 			},
 			"dataset": schema.StringAttribute{
-				Required:    true,
+				Optional:    true,
+				Computed:    true,
 				Description: "The dataset this Burn Alert is associated with.",
 				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.RequiresReplace(),
+					modifiers.DatasetDeprecation(),
+					stringplanmodifier.RequiresReplaceIf(
+						func(ctx context.Context, sr planmodifier.StringRequest, rrifr *stringplanmodifier.RequiresReplaceIfFuncResponse) {
+							// If the dataset is empty, we don't want to require a replace
```

Comment:
running the test suite with coverage shows that this `RequiresReplaceIf` function isn't being executed at all 🤔 

Are we sure we need the function, or is the test not sufficient?